### PR TITLE
Add Result Filters to Pages

### PR DESCRIPTION
**Pull request by rynowak**
Look at the commits to get a good sense of what's going on here.

The main thrust is to reuse as many of the tests for controllers as possible

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/ResourceInvoker.cs [hunk 1]
```diff
@@ -447,11 +443,419 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                     }
 
                 case State.ResourceInside:
+                {
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/ResourceInvoker.cs]: Presumably this is just stuff moved from the other invoker. I don't need to fine-grain review it, right?

**Comment by pranavkm** [comment]
LKGTM :shipit:

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/ResourceInvoker.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/ResourceInvoker.cs]: Yes, it was mostly copy-paste and redoing what you undid before split it up.

**Pull request merged**